src/ir/analysis/derive_partial_eq.rs

```diff
@@ -144,6 +145,11 @@ impl<'ctx> MonotoneFramework for CannotDerivePartialEq<'ctx> {
             };
         }
 
+        let name = item.canonical_name(self.ctx);
+        if self.ctx.options().no_partialeq_types.matches(&name) {
```

Comment:
Let's abstract this check into a `BindgenContext::no_partialeq_by_name`, similar to `BindgenContext::is_opaque_by_name` and `BindgenContext::is_blacklisted_by_name`.